### DISCUSSION TITLE
Custom gunicorn configuration file for logging

### DISCUSSION THREAD

**Discussion by thomascenni** [tiangolo/uvicorn-gunicorn-docker#230] [Questions] [answered]
Hi,
I have a custom gunicorn file with:

...
access_log_format = '%({x-forwarded-for}i)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"'
...

This custom gunicorn file was working perfectly with the corresponding wsgi image at https://github.com/tiangolo/meinheld-gunicorn-docker.

When I use the image at https://github.com/tiangolo/uvicorn-gunicorn-docker, I just changed my Django application MODULE_NAME from the wsgi to the asgi. But the logs are without the information requested on access_log_format.

I really don't understand, I am struggling many hours with this; is it something related to uvicorn logging that is not "passing back" to gunicorn?

**Comment by tiangolo** [comment, marked as answer]
Now that Uvicorn supports managing workers with `--workers`, including restarting dead ones, there's no need for Gunicorn. That also means that it's much simpler to build a Docker image from scratch now, I updated the docs to explain it.

Because of that, I deprecated this Docker image: https://github.com/tiangolo/uvicorn-gunicorn-docker#-warning-you-probably-dont-need-this-docker-image

That also means that configuring logs would be much simpler as you would only need to configure Uvicorn, not Gunicorn no top of Uvicorn.
